portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -26,36 +29,39 @@
 /** Servlet that handles comments data */
 @WebServlet("/comments")
 public class CommentsServlet extends HttpServlet {
+  private static final String JSON_CONTENT_TYPE = "application/json;";
+  private static final String COMMENTS_PAGE_URL = "/comments-page.html";
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  CommentDatabase commentDatabase = new CommentDatabase(datastore);
 
-  private ArrayList<Comment> comments;
-  
-  @Override
-  public void init() {
-    comments = new ArrayList<Comment>();
-  }
-  
   /* responds with a JSON string containing authors and comments*/
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Query datastore for all comments
+    ArrayList<Comment> comments = commentDatabase.getCommentsAsArrayList();
     // Convert the ArrayLists to JSON
-
     Gson gson = new Gson();
     String commentsJson = gson.toJson(comments);
     // Send the JSON as the response
-    response.setContentType("application/json;");
+    response.setContentType(JSON_CONTENT_TYPE);
     response.getWriter().println(commentsJson);
   }
 
   /* when the submit button on the values page is hit, doPost requests the text in the author and commment field and stores them*/ 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the comments form
-    String authorText = getParameter(request, "author");
-    String valueText = getParameter(request, "value");
-    comments.add(new Comment(authorText, valueText));
-
+    try {
+      String authorText = getParameter(request, commentDatabase.AUTHOR_FIELD);
+      String valueText = getParameter(request, commentDatabase.VALUE_FIELD);
+      Comment comment = new Comment(authorText, valueText);
+      commentDatabase.putCommentInDatabase(comment);   
+    } catch (RuntimeException e){ 
+        e.printStackTrace();
+        System.out.println("Comment entered is not a valid value.");
```

Comment:
Do you plan on later redirecting the user to an error page? Right now you just redirect them back to the comments page with no indication why their comment isn't posted, right?

Reply:
I was thinking trying to pop up an html alert but I like the idea of an error page.